presto-orc/src/main/java/com/facebook/presto/orc/reader/SliceDictionarySelectiveReader.java

```diff
@@ -0,0 +1,602 @@
+                int succeedingPositionsToFail = filter.getSucceedingPositionsToFail();
+                if (succeedingPositionsToFail > 0) {
+                    int positionsToSkip = 0;
+                    for (int j = 0; j < succeedingPositionsToFail; j++) {
+                        i++;
+                        int nextPosition = positions[i];
+                        positionsToSkip += 1 + nextPosition - streamPosition;
+                        streamPosition = nextPosition + 1;
+                    }
+                    skip(positionsToSkip);
+                }
+            }
+        }
+        return streamPosition;
+    }
+
+    private boolean isCharType()
+    {
+        return streamDescriptor.getOrcType().getOrcTypeKind() == CHAR;
+    }
+
+    private int readAllNulls(int[] positions, int positionCount)
+            throws IOException
+    {
+        presentStream.skip(positions[positionCount - 1]);
+
+        if (nonDeterministicFilter) {
+            outputPositionCount = 0;
+            for (int i = 0; i < positionCount; i++) {
+                if (filter.testNull()) {
+                    outputPositionCount++;
+                }
+                else {
+                    outputPositionCount -= filter.getPrecedingPositionsToFail();
+                    i += filter.getSucceedingPositionsToFail();
+                }
+            }
+        }
+        else if (nullsAllowed) {
+            outputPositionCount = positionCount;
+        }
+        else {
+            outputPositionCount = 0;
+        }
+
+        allNulls = true;
+        return positions[positionCount - 1] + 1;
+    }
+
+    private void ensureValuesCapacity(int capacity)
+    {
+        values = ensureCapacity(values, capacity);
+    }
+
+    private void skip(int items)
+            throws IOException
+    {
+        if (presentStream != null) {
+            int dataToSkip = presentStream.countBitsSet(items);
+            if (inDictionaryStream != null) {
+                inDictionaryStream.skip(dataToSkip);
+            }
+            if (dataStream != null) {
+                dataStream.skip(dataToSkip);
+            }
+        }
+        else {
+            if (inDictionaryStream != null) {
+                inDictionaryStream.skip(items);
+            }
+            dataStream.skip(items);
+        }
+    }
+
+    @Override
+    public int[] getReadPositions()
+    {
+        return outputPositions;
+    }
+
+    @Override
+    public Block getBlock(int[] positions, int positionCount)
+    {
+        checkArgument(outputPositionCount > 0, "outputPositionCount must be greater than zero");
+        checkState(outputRequired, "This stream reader doesn't produce output");
+        checkState(positionCount <= outputPositionCount, "Not enough values");
+
+        if (allNulls) {
+            return new RunLengthEncodedBlock(new DictionaryBlock(1, dictionaryBlock, new int[] {dictionaryBlock.getPositionCount() - 1}), positionCount);
```

Comment:
this can be simplified to 

`return new RunLengthEncodedBlock(outputType.createBlockBuilder(null, 1).appendNull().build(), outputPositionCount);`